resources/patches/ruby_251/windows_ruby_2.5_fixup_generated_batch_files.patch

```diff
@@ -0,0 +1,86 @@
+   ruby.exe in the same working directory as the binary shims.  This
+   worked as desired to ensure that the vendored gems / binary shims
+   shipped with Puppet on Windows would always use the vendored Ruby
+   that we ship.
+
+   However, we've recently created a need in our CI environment to also
+   use the Ruby version that we build custom / vendor (partially due
+   to our custom Ruby supporting Windows 2003 with Ruby 2.1.5 whereas
+   the RubyInstaller version does not).  In the spec running environment
+   Pik is used to select a Ruby and inject it into PATH.  When gems are
+   installed via bundle install --path .bundle/gems, the expectation is
+   that the Ruby being used is the Pik'd version.  In other words, the
+   use case is slightly different from the vendored Ruby shipped and
+   packaged in the MSI.  Therefore, the batch file generation has been
+   modified to first favor a ruby.exe in the current working directory,
+   but to fallback to using PATH otherwise.
+---
+ lib/rubygems/commands/setup_command.rb | 13 +++++++++----
+ lib/rubygems/installer.rb              | 13 +++++++++----
+ 2 files changed, 18 insertions(+), 8 deletions(-)
+
+diff --git a/lib/rubygems/commands/setup_command.rb b/lib/rubygems/commands/setup_command.rb
+index ee6ce88eae..2cda5866de 100644
+--- a/lib/rubygems/commands/setup_command.rb
++++ b/lib/rubygems/commands/setup_command.rb
+@@ -256,14 +256,19 @@ def install_executables(bin_dir)
+             bin_cmd_file = File.join Dir.tmpdir, "#{bin_file}.bat"
+ 
+             File.open bin_cmd_file, 'w' do |file|
+-              file.puts <<-TEXT
++              file.puts <<-SCRIPT
+   @ECHO OFF
+   IF NOT "%~f0" == "~f0" GOTO :WinNT
+-  @"#{File.basename(Gem.ruby).chomp('"')}" "#{dest_file}" %1 %2 %3 %4 %5 %6 %7 %8 %9
++  ECHO.This version of Ruby has not been built with support for Windows 95/98/Me.
+   GOTO :EOF
+   :WinNT
+-  @"#{File.basename(Gem.ruby).chomp('"')}" "%~dpn0" %*
+-  TEXT
++  IF EXIST "%~dp0ruby.exe" (
++  SET RUBY_EXE_PATH="%~dp0ruby.exe"
++  ) ELSE (
++    SET RUBY_EXE_PATH="ruby.exe"
++  )
++  @%RUBY_EXE_PATH% "%~dpn0" %*
++  SCRIPT
+             end
+ 
+             install bin_cmd_file, "#{dest_file}.bat", :mode => 0755
+diff --git a/lib/rubygems/installer.rb b/lib/rubygems/installer.rb
+index ee5fedeb64..e0a228f9b5 100644
+--- a/lib/rubygems/installer.rb
++++ b/lib/rubygems/installer.rb
+@@ -754,14 +754,19 @@ def app_script_text(bin_file_name)
+ 
+   def windows_stub_script(bindir, bin_file_name)
+     ruby = Gem.ruby.gsub(/^\"|\"$/, "").tr(File::SEPARATOR, "\\")
+-    return <<-TEXT
++    return <<-SCRIPT
+ @ECHO OFF
+ IF NOT "%~f0" == "~f0" GOTO :WinNT
+-@"#{ruby}" "#{File.join(bindir, bin_file_name)}" %1 %2 %3 %4 %5 %6 %7 %8 %9
++ECHO.This version of Ruby has not been built with support for Windows 95/98/Me.
+ GOTO :EOF
+ :WinNT
+-@"#{ruby}" "%~dpn0" %*
+-TEXT
++IF EXIST "%~dp0ruby.exe" (
++  SET RUBY_EXE_PATH="%~dp0ruby.exe"
++) ELSE (
++  SET RUBY_EXE_PATH="ruby.exe"
```

Comment:
I'd really prefer to remove this ELSE case and update the description in the patch to match why we actually need it now, instead of just taking this as-is from what was done in the past. There should be no reason we'd ever need to fall back to another ruby here, and I agree with @McdonaldSeanp that it could potentially cause problems.

If PDK relies on this bit of the behavior, I'd like to better understand why/how so we can maybe not do that.

Otherwise I think this is fine to land

Reply:
Yeah, that would work for me.

Reply:
Yeah, should work.